src/code/z_camera.c

```diff
@@ -1,42 +1,243 @@
 #include "global.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_fabsf.s")
+/**
+ * Returns the absolute value for floats
+ */
+f32 Camera_fabsf(f32 f) {
+    return ABS(f);
+}
+
+/**
+ * Returns the magnitude for 3D float vectors
+ */
+f32 Camera_Vec3fMagnitude(Vec3f* vec) {
+    return sqrtf(SQ(vec->x) + SQ(vec->y) + SQ(vec->z));
+}
+
+/**
+ * Interpolates along the curve shown below
+ * returns value y ranging from 0.0f to 1.0f for -xMax <= x <= xMax
+ * returns 1.0f otherwise
+ *
+ * y = 1.0f    ________                   _________
+ *                     __               __
+ *                       _             _
+ * y axis                 _           _
+ *                         ___     ___
+ *                            _____
+ * y = 0.0f           |         |         |
+ *                 -xMax        0       xMax
+ *
+ *                           x axis
```

Comment:
I'm surprised, what's not clear about? It's a curve on the x-y plane

Reply:
Yes I can see it's a function, but the fidelity's never going to be satisfactory for this sort of thing. What it is really is a C^1 interpolation between (0,0) with derivative 0 and (R,1) with derivative 0, and the crossover point chosen as at 0.6R, not sure why they chose it instead of 0.4 or 0.5.
<img width="610" alt="image" src="https://user-images.githubusercontent.com/73679967/184654081-959057a5-b8b2-41ef-8334-a3bae36a138b.png">